core/src/validator.rs

```diff
@@ -77,18 +77,33 @@ impl Validator {
         authority: &AccountId,
         transaction: VersionedSignedTransaction,
     ) -> Result<(), ValidationFail> {
-        let runtime = wasm::RuntimeBuilder::<wasm::state::validator::ValidateTransaction>::new()
-            .with_engine(wsv.engine.clone()) // Cloning engine is cheap, see [`wasmtime::Engine`] docs
-            .with_configuration(wsv.config.wasm_runtime_config)
-            .build()?;
-
         trace!("Running transaction validation");
-        runtime.execute_validator_validate_transaction(
-            wsv,
-            authority,
-            &self.loaded_validator.module,
-            transaction,
-        )?
+
+        match self {
+            Self::Initial => {
+                let (_authority, Executable::Instructions(instructions)) = transaction.into() else {
+                    return Ok(());
+                };
+                for isi in instructions {
+                    self.validate_instruction(wsv, authority, isi)?;
```

Comment:
```suggestion
                    isi.execute(authority, wsv).map_err(Into::into)?;
```

I would rather have it like this